pandas/io/json/_json.py

```diff
@@ -726,7 +740,10 @@ def read(self):
             obj = concat(self)
         elif self.lines:
             data = ensure_str(self.data)
-            obj = self._get_object_parser(self._combine_lines(data.split("\n")))
+            data = data.split("\n")
+            if self.nrows:
```

Comment:
Can you implement this at a lower level? I think should be done in the C extension to match behavior of other IO methods

Reply:
I have not worked with lower-level extension till now. If it's to be done there, I will look into it
Also, it would be very nice to get some direction on how to achieve it.

Reply:
I actually don't think we need to implement this on a lower level  (lines is not implemented that way currently).

rather you need to slice this differently to avoid putting all of this into memory

e.g. you can do this: https://stackoverflow.com/questions/3862010/is-there-a-generator-version-of-string-split-in-python

to create a generator on the split rows, then islice them, process and repeat